lib/ClangImporter/ClangImporter.cpp

```diff
@@ -5080,31 +5082,41 @@ ClangTypeEscapability::evaluate(Evaluator &evaluator,
         importer::getConditionalEscapableAttrParams(recordDecl);
     if (!conditionalParams.empty()) {
       auto specDecl = cast<clang::ClassTemplateSpecializationDecl>(recordDecl);
-      auto templateDecl = specDecl->getSpecializedTemplate();
       SmallVector<std::pair<unsigned, StringRef>, 4> argumentsToCheck;
-      for (auto [idx, param] :
-           llvm::enumerate(*templateDecl->getTemplateParameters())) {
-        if (conditionalParams.erase(param->getName()))
-          argumentsToCheck.push_back(std::make_pair(idx, param->getName()));
-      }
       HeaderLoc loc{recordDecl->getLocation()};
-      for (auto name : conditionalParams)
-        desc.impl.diagnose(loc, diag::unknown_template_parameter, name);
-
-      auto &argList = specDecl->getTemplateArgs();
-      for (auto argToCheck : argumentsToCheck) {
-        auto arg = argList[argToCheck.first];
-        if (arg.getKind() != clang::TemplateArgument::Type) {
-          desc.impl.diagnose(loc, diag::type_template_parameter_expected,
-                             argToCheck.second);
-          return CxxEscapability::Unknown;
+      while (specDecl) {
+        auto templateDecl = specDecl->getSpecializedTemplate();
+        for (auto [idx, param] :
+             llvm::enumerate(*templateDecl->getTemplateParameters())) {
+          if (conditionalParams.erase(param->getName()))
+            argumentsToCheck.push_back(std::make_pair(idx, param->getName()));
         }
+        auto &argList = specDecl->getTemplateArgs();
+        for (auto argToCheck : argumentsToCheck) {
+          auto arg = argList[argToCheck.first];
+          if (arg.getKind() != clang::TemplateArgument::Type) {
+            desc.impl.diagnose(loc, diag::type_template_parameter_expected,
+                               argToCheck.second);
+            return CxxEscapability::Unknown;
+          }
 
-        auto argEscapability = evaluateEscapability(
-            arg.getAsType()->getUnqualifiedDesugaredType());
-        if (argEscapability == CxxEscapability::NonEscapable)
-          return CxxEscapability::NonEscapable;
+          auto argEscapability = evaluateEscapability(
+              arg.getAsType()->getUnqualifiedDesugaredType());
+          if (argEscapability == CxxEscapability::NonEscapable)
+            return CxxEscapability::NonEscapable;
+        }
+        clang::DeclContext *rec = specDecl;
+        specDecl = nullptr;
+        while ((rec = rec->getParent())) {
```

Comment:
Should this loop stop once `rec->isTranslationUnit()`?

Reply:
When `rec->isTranslationUnit()` is true, `rec->getParent()` returns null and the loop terminates. That being said, I just realized `rec` is a bit confusing name here.

Reply:
I see, that makes sense.